### PR TITLE
fix(code-review): resolve PR base host-agnostically (GitHub Enterprise fork fix)

### PR DESCRIPTION
**Pull request by davidalee**
## Summary

The stable `ce-code-review` skill silently picks the wrong merge base — or no base at all — on **GitHub Enterprise** or any non-`github.com` host when the reviewer is working from a fork. The base-resolution awk block in `SKILL.md` matches remotes by literal `github.com:` / `github.com/` substring, and the HTTPS fallback URL is hard-coded to `https://github.com/...`. On GHE, `PR_BASE_REMOTE` ends up empty, the script falls through to `origin`, and in fork workflows `origin` is the user's fork — so the resulting diff is either empty, incorrect, or relative to the wrong base.

This PR makes base resolution host-agnostic: the agent derives both `<base-host>` and `<base-repo>` from the same PR URL it already parses, and the awk block extracts `(host, owner/repo)` from each remote URL generically. No behavior change for `github.com` users.

## How this was uncovered

While building **`ce-code-review-beta`** in #784 (Codex CLI delegation), the Codex code-review bot flagged the identical hard-coding pattern in the beta variant's `resolve-base.sh`:

> The PR-base detection is hard-coded to `https://github.com/...` and later remote matching is hard-coded to `github.com:` / `github.com/`, so reviews on GitHub Enterprise (or any non-`github.com` host) will fail to identify the true base remote. In fork setups this can silently fall back to `origin`, producing the wrong merge base (or no base) even when the correct upstream remote exists.

That feedback was resolved in #784. Auditing the stable skill on `main` confirmed the same root cause — now living inline in `SKILL.md` after #815 replaced the standalone script with prose-driven base detection. This PR ports the fix.

## What changed

`plugins/compound-engineering/skills/ce-code-review/SKILL.md` — base-resolution prose and awk:

- **Prose:** instructs the agent to derive both `<base-host>` and `<base-repo>` from the PR URL (previously only the repo portion). Calls out case-insensitivity and shows a GHE example alongside the existing `github.com` example.
- **awk:** replaces the two `index($2, "github.com:<base-repo>") || index($2, "github.com/<base-repo>")` substring checks with a per-remote parser that handles:
  - `https://[user@]host[:port]/owner/repo[.git]`
  - `ssh://[user@]host[:port]/owner/repo[.git]`
  - scp-form `user@host:owner/repo[.git]`
  - Strips userinfo, port, and trailing `.git`; lowercases host and owner/repo before exact-equality comparison.
- **HTTPS fallback URL:** `https://github.com/<base-repo>.git` → `https://<base-host>/<base-repo>.git`.

## Why this is safe to merge

- **No new dependencies, no new scripts**: the change stays inline per #815's design intent (prose-driven base detection).
- **No behavior change on `github.com`**: the new awk produces the same remote pick for canonical `github.com` URLs, verified across HTTPS, scp-form, mixed-case, userinfo, port, and `.git`/no-`.git` variants.
- **Boundary case preserved**: `org/repo` PR does not spuriously match a remote at `org/repo-extra` (exact equality, not substring).
- **No test regressions**: full `bun test` suite passes (1340/1340); `bun run release:validate` clean.
- **Diff is small**: 1 file, +15/-4 lines.

## Test plan

- [x] `bun test` — all 1340 tests pass on this branch
- [x] `bun run release:validate` — metadata in sync
- [x] Awk smoke test with realistic remote URL forms:
  - `git@ghe.acme.com:EveryInc/compound-engineering-plugin.git` matches `(ghe.acme.com, everyinc/compound-engineering-plugin)` ✓
  - `https://x-token@ghe.acme.com/Org/Repo.git` matches `(ghe.acme.com, org/repo)` ✓
  - `git@github.com:org/repo-extra.git` does **not** match `(github.com, org/repo)` ✓
  - `git@github.com:org/repo.git` still matches `(github.com, org/repo)` ✓ (regression guard)

## Related

- Upstream feedback that surfaced this: [Codex review on #784](https://github.com/EveryInc/compound-engineering-plugin/pull/784#issuecomment-4390521308)
- Fix in beta variant: davidalee/compound-engineering-plugin@c09b90d7 (part of PR #784)
- Prior design context: #815 (replaced the standalone `resolve-base.sh` with inline prose)

### CODE REVIEW

File: plugins/compound-engineering/skills/ce-code-review/SKILL.md [hunk 1]
```diff
@@ -246,20 +246,31 @@ Then fetch PR metadata. Capture the base branch name and the PR base repository
...
+  {
+    url = $2; h = ""; p = ""
+    if (match(url, /:\/\/[^\/]+\//))    { h = substr(url, RSTART+3, RLENGTH-4); p = substr(url, RSTART+RLENGTH) }
+    else if (match(url, /@[^:]+:/))     { h = substr(url, RSTART+1, RLENGTH-2); p = substr(url, RSTART+RLENGTH) }
+    else next
+    sub(/^[^@]*@/, "", h); sub(/:[0-9]+$/, "", h)
+    h = tolower(h)
+    if (!match(p, /^[^\/]+\/[^\/]+/)) next
+    pr = substr(p, 1, RLENGTH); sub(/\.git$/, "", pr); pr = tolower(pr)
+    if (h == host && pr == repo) { print $1; exit }
```
[the remark this answers is not in the document]
Comment (davidalee) [reply plugins/compound-engineering/skills/ce-code-review/SKILL.md]: > The new remote matcher strips `:<port>` from each remote host (`h`) but compares it to `<base-host>` unchanged...

Addressed: the matcher now preserves ports for URL-form remotes, explicitly documents ported PR hosts, and allows scp-form SSH remotes to match the same host without the web UI port. I also added an executable regression test covering HTTPS, ssh URL, scp-form fallback, and wrong-port rejection cases.
